### PR TITLE
Lingo: Add pilgrimage logic through Starting Room

### PR DESCRIPTION
**Pull request by hatkirby**
## What is this fixing or adding?
The Starting Room previously did not contain entrances from adjacent rooms (only entrances from arrow paintings). In regular logic, this does not matter, because you always have access to the Starting Room. However, when pilgrimage is enabled, the generator has to route a warpless path between successive sunwarps, which may involve going back to the Starting Room and exiting through one of the other doors. This adds those connections.

## How was this tested?
pytest, including a new unit test specifically to test this issue.

### CODE REVIEW

**Review by Exempt-Medic — APPROVED** [review]
Changes LGTM, the unit test passes and this matches my understanding of the game (all four entrances are accounted for and the scenario requires the proper navigation)

My offline comment about the unit test ordering was also addressed

File: worlds/lingo/test/TestPilgrimage.py [hunk 1]
```diff
@@ -88,6 +88,35 @@ def test_access(self):
...
+                 "Starting Room - Street Painting", "Outside The Initiated - Shortcut to Hub Room",
+                 "Directional Gallery - Shortcut to The Undeterred", "Orange Tower First Floor - Salt Pepper Door",
+                 "Color Hunt - Shortcut to The Steady", "The Bearer - Entrance",
+                 "Orange Tower Fifth Floor - Quadruple Intersection", "The Tenacious - Shortcut to Hub Room",
+                 "Outside The Agreeable - Tenacious Entrance", "Crossroads - Tower Entrance",
+                 "Orange Tower Fourth Floor - Hot Crusts Door", "Challenge Room - Welcome Door",
+                 "Number Hunt - Challenge Entrance", "Welcome Back Area - Shortcut to Starting Room"]
+
+        for door in doors:
+            print(door)
```
Comment (black-sliver) [inline worlds/lingo/test/TestPilgrimage.py:113]: While maybe nice when debugging the test, that print should probably not be here. If you want to get a proper message (on failure or verbose mode), please use the 2nd argument to `self.assertFalse()` or put the assert into an `with self.subTest(door=door):`

**Review by black-sliver — COMMENTED** [review]
Don't know Lingo well enough to know anything about those connections, but a minor nit-pick

File: worlds/lingo/test/TestPilgrimage.py (hunk 1 above)
Comment (hatkirby) [reply worlds/lingo/test/TestPilgrimage.py]: Whoops, that was mistakenly left in from debugging. Fixed.

**Comment by black-sliver** [comment]
> Update generated.dat

While out of scope of this PR, I think adding a test that validates generated.dat against the yaml would be neat

**Comment by hatkirby** [comment]
> > Update generated.dat
> 
> While out of scope of this PR, I think adding a test that validates generated.dat against the yaml would be neat

There is one already. I just had to update the data file after merging in the changes from the other PR.

**Comment by black-sliver** [comment]
Yeah, I realized that right after pressing enter and checked. Sorry for the noise. :S

**Review by black-sliver — APPROVED** [review]

**Pull request merged**